### ISSUE TITLE
Global navigation: Handle too many items

### ISSUE THREAD

**Issue by benoitf** [containers/podman-desktop#6774]
### Is your enhancement related to a problem? Please describe

the navbar contains too many items for the default min-height that we have

if there are too many items to be displayed, buddy icon and settings icon should always be there but we should hide the previous icons with a `...` and with either our custom dropdown (like Slack for example) or a menu (like VS Code). I would opt-in for dropdown but it looks like there was a problem when wanting to use it for the buddy dropdown

To mitigate the issue we could reduce the vertical gap in each icon of the navbar from py-3 to py-2

### Describe the solution you'd like

quick & dirty: reduce py-3 to py-2, reduce icon size
proper handle: hide top items and display ... / more options while bottom icons should always be visible (never hidden)

**Comment by deboer-tim** [comment]
Since it's an IDE I typically use a near full-screen window for VS Code, and it has 8 nav items without any extensions.
I typically use Podman Desktop at 60% of that size, and we'll now have 11 in the default configuration. Without changes the first installed extension would be visible, the second would be hidden.

We should absolutely do the ..., py-2, etc., and that will squeeze a couple more in and make it work better for now, but IMHO our core problem is that we have too many items in the nav and keep adding more. It looks cluttered, it's time to remove a few options or look at other options.

**Comment by benoitf** [comment]
commented there https://github.com/containers/podman-desktop/pull/6777#pullrequestreview-2006945794 that we could also reduce the width of the navbar

**Comment by ekidneyrh** [comment]
Just adding my thoughts here ahead of the UX call today. 

Since we have modelled the nav bar from VS code, we can have a look and see what they are doing: 

![Screenshot from 2024-06-20 12-03-33](https://github.com/containers/podman-desktop/assets/87311656/d58d8b40-24b2-4b2b-a6f9-e8684de9dad3)

You can right-click an empty space on the nav bar and then select what icons you want pinned in the nav bar. If right-clicked over an icon, an additional option to ‘unpin’ will appear.

![Screenshot from 2024-06-20 13-57-50](https://github.com/containers/podman-desktop/assets/87311656/f321c1f9-6ecf-4911-8afa-e74a94e3fa3c)


Benefit of us doing the same: developers using podman will notice the similarity in VS Code and will know to right-click to do this. 

Downsides would be users that are not familiar with this layout might not know how. We can work around this by also having the option to pin/unpin to the nav bar through the extensions tab. 
	
We need to decide what items should be there on install / by default. What is podman for initially without the extra features? What are it’s basic functionalities? We can keep this to 6 or 7 as that’s the amount of information a user can hold in their short term memory. 

I think what I have in the mock-up above works: 

- Dashboard
- Containers
- Pods
- Images
- Volumes
- Extensions
- Settings 

The user can then add in new extensions as they normally would, and then it would appear in the nav bar. I would get rid of the inset for this route as it would get too complicated. Just have them slot in at the end of the row. Set a max amount of pinned items to 10? 

For the coloured icons, we can get monotones ones so they fit in with the rest of the icons.

**Comment by deboer-tim** [comment]
Some thoughts:
- At minimum height we can maybe get 10 items in, but if you have a taller window you could do more. Instead of a specific limit, maybe we do the same thing as VS Code and use a ... menu for anything that doesn't fit?
- Kubernetes is 5 items and 1 more on the way. I think you're suggesting getting rid of nav groups above (which I'm good with), but to me it feels like you wouldn't want to automatically add all 6 by default nor individually add/remove. To me it feels like we're going to have more items in general than VS Code and need something a little more to find & manage them.
- For the last item, maybe a way to add items to a secondary or 'not as frequently used' menu? e.g. Other Kubernetes.
- IMHO extensions like AI Lab should be able to contribute multiple nav items instead of one. This would allow it to be 'first class' in the nav and take advantage of everything above, but also adds to scale issues.

**Comment by ekidneyrh** [comment]
The max number of item was a suggestion, so I would be good with having anything extra under a more menu etc.

For the Kubernetes stuff, could they be grouped together within the catalog? Would there be a use case for all of them to be downloaded and added together, or can they be separated? 

I think a 'not as frequently' used menu could be that more menu mentioned above with the items that down fit in the nav bar. Have the user be able to re-arrange the items in a preferred order, so the less used one can be grouped in such menu. 

WDYM by multiple nav items? Like how Kubernetes has multiple items? I think having that within the AI Lab extension is best, as it has its own navigation menu within it and keeps it contained there. This could also be done for Kubernetes - have an additional navigation menu within that item and keep them contained and grouped together that way.

**Comment by axel7083** [comment]
Here is some examples from Jetbrain navigation system. They recently (1-2 years ago) started a very big revamp of their UI to simplify it, and improve the way developers uses it.

### Submenus

![image](https://github.com/containers/podman-desktop/assets/42176370/2c3bd281-d292-43df-9dba-228087066d56)

> I really like the keyboard shortcuts

### Moving icons

![image](https://github.com/containers/podman-desktop/assets/42176370/9e01474f-f57f-4d91-8d06-ef8c4d29e924)

They are a separation in the navigation bar, and you can move those item

### Right click on an icon in the nav bar

![image](https://github.com/containers/podman-desktop/assets/42176370/c45f3a72-f154-49c4-b499-bf18178a3579)

### When `Show Tool Window Names` is enable

![image](https://github.com/containers/podman-desktop/assets/42176370/f0915ff1-168d-4725-ad30-72487d5c3d17)

### Top and bottom items

![image](https://github.com/containers/podman-desktop/assets/42176370/645aa3fc-5326-4d2a-a4b6-47f26811809a)

### Scalable

https://github.com/containers/podman-desktop/assets/42176370/dd174657-c603-40b3-85ea-f14fd4f36514

### Recent item 

When an item, which is in the `...` is selected, it get on the navbar, until a new one is opened

**Comment by ekidneyrh** [comment]
Discussed this during our UX call 20/06/2024

### Have the ability to customise what icons are on the main nav.
The user can pin them as I've done in the mock-up below:
 
> You can right-click an empty space on the nav bar and then select what icons you want pinned in the nav bar. If right-clicked over an icon, an additional option to ‘unpin’ will appear.
> 
> ![Screenshot from 2024-06-20 13-57-50](https://private-user-images.githubusercontent.com/87311656/341420754-f321c1f9-6ecf-4911-8afa-e74a94e3fa3c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyMTk2NDgsIm5iZiI6MTcxOTIxOTM0OCwicGF0aCI6Ii84NzMxMTY1Ni8zNDE0MjA3NTQtZjMyMWMxZjktNmVjZi00OTExLThhZmEtZTc0YTk0ZTNmYTNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI0VDA4NTU0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1YzRlNTk5Y2EyMGYyODdiMzk4ZThiZTU5MTIxNzRhNTViZWFhZWVjMzJjNjBlZTU4ZjRiZjgxMTZlYTg1OTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8teD4N1-MPrLrnOsSqAfeQ6BeQjRGlwWd4AXT4tDBok)
>

### For the overflow of items, have them below a '...' menu 
such as the example Axel gave above:

![image](https://github.com/containers/podman-desktop/assets/87311656/0cc6488d-15d6-4dbc-8812-05e9911abeba)

###  Get rid of nav groups and have a single nav group item instead 
with a submenu that follow the pattern of AI Lab. 

![image](https://github.com/containers/podman-desktop/assets/87311656/04d226e8-b0ca-4bea-b3ea-1fe39897ce10)

If the user wants they can add an item from the submenu to the main nav eg Deployments from Kubernetes. This can be done through a similar process as above; right click and have the option to pin. 

### Icons from extensions to match monotone style of default icons 
Speaks for itself, have the icons be in one colour to not look too different from the others.

**Comment by axel7083** [comment]
still needed

**Comment by vancura** [comment]
This issue is being addressed as part of the UI chrome redesign (#12709).

**Comment by MariaLeonova** [comment]
@slemeur and I have been discussing the fact that the bottom menu items disappear once you have many extensions and resize the window vertically.
Can we investigate the option of keeping Accounts and Settings sticky at the bottom? And potentially adding a scroll functionality to the nav.

**Comment by MariaLeonova** [comment]
Some examples of long list in the left nav UI: Figma, Penpot, Slack.

**Comment by vancura** [comment]
I am not sure scrolling would help. From top of my head, there are two options we have:

1. either we collapse icons to be vertically smaller when there's no space,
2. or bundle icons together in groups (similarly to what Discord does in the sidebar).

I am afraid scrolling won't help much, as it would always make some icons offscreen. Well, that's what would happen with 2. as well.

Collapsing (making the icons smaller) won't help on the long run either, we'd just delay the problem again, we'd just buy some time (by buying space).

**Comment by vzhukovs** [comment]
> I am not sure scrolling would help. From top of my head, there are two options we have:
> 
> 1. either we collapse icons to be vertically smaller when there's no space,
> 2. or bundle icons together in groups (similarly to what Discord does in the sidebar).
> 
> I am afraid scrolling won't help much, as it would always make some icons offscreen. Well, that's what would happen with 2. as well.
> 
> Collapsing (making the icons smaller) won't help on the long run either, we'd just delay the problem again, we'd just buy some time (by buying space).

Provided an interim solution for the first iteration: 

https://github.com/podman-desktop/podman-desktop/pull/13008
https://github.com/podman-desktop/podman-desktop/pull/13009

on the next iteration, when we will have a clear view how to render extension list, we can refactor the navigation bar.

**Comment by vancura** [comment]
## Prototype available for review!

I've built a fully functional prototype that addresses this issue with a comprehensive solution: https://github.com/vancura/podman-desktop-navbar-proto (hosted version at https://podman-desktop-navbar-proto.vancura.dev).

### What it demonstrates:

- **Four-tier structure**: Essentials, Pinned, Regular items, and always-visible Bottom panel (Account + Settings)
- **"More" button**: Hidden items accessible via dropdown, exactly as discussed in this issue.
- **Resizable navbar**: Drag to switch between icon-only and icon+title modes
- **Pinning system**: Users can pin frequently-used items (VSCode/Teams-style)
- **Context menus**: Right-click items for pinning, hiding, keyboard shortcuts, and more
- **Full i18n + RTL**: Tested with five languages, including Arabic/Hebrew

This is a functional prototype to validate UX patterns before high-fidelity design. The prototype showcases [Phase 1](https://github.com/vancura/podman-desktop-navbar-proto#phase-1-core-navigation-current), and there's a roadmap for [Phase 2](https://github.com/vancura/podman-desktop-navbar-proto#phase-2-drag--drop-planned) (drag & drop) and [Phase 3](https://github.com/vancura/podman-desktop-navbar-proto#phase-3-workspaces--profiles-future) (workspace profiles).

### Quick tips to explore:

- **Hold ⌘ (or CTRL)** to reveal keyboard shortcuts above items
- **Right-click** anywhere in the navbar for context menus
- **Drag the edge** to resize and watch modes switch automatically.

### Feedback welcome:

We're looking for feedback on **interaction patterns and UX**, not the visual design (colors, spacing, icons are placeholders).

- Does the four-tier structure make sense?
- Is pinning/hiding intuitive?
- Does the More button solve the overflow problem effectively?
- Any edge cases or workflows we're missing?

**Comment by axel7083** [comment]
Thanks for sharing your work on the navbar !

> Any edge cases or workflows we're missing?

I am repeating some exchange we had in private to have a public trace:

I think the title of the items should be visible even in minimized version, we had the feedback in the past that the icon alone was not enough for new users, and being able to see the name directly without being it force to expand the menu and reducing the main windows is helpful. (current default behavior is showing the titles under the icons).

<img width="182" height="88" alt="Image" src="https://github.com/user-attachments/assets/8aed0aa9-58fe-428c-b83b-1eaf4b6ea423" />

**Comment by deboer-tim** [comment]
I agree, but we already have a preference for this: Navigation Bar Layout. Titles are on by default but user can change to icons-only once they're familiar with them.

**Comment by slemeur** [comment]
@axel7083: is the use case you are pointing out the one primarily of a new user? I believe, in the case of a new user, the navbar will not be collapsed but expanded, is that what you are intending @vancura ?  
In this case, that gives the opportunity for the new users to get used to the icons, before they decide to collapse the navbar (which would be their choice to do it or not). 

If at some point, they forget the meaning of the icon, there are tooltips.

**Comment by vancura** [comment]
Yes, exactly as what @slemeur says: new users (i.e., defaults) would see the wider version with both the label and the icon.

**Comment by ScrewTSW** [comment]
Still present in 1.25.0, possibly regression? Having too many extensions installed causes the bar to be unusable, cannot reach Settings or Account sections, no scroll functionality

<img width="111" height="712" alt="Image" src="https://github.com/user-attachments/assets/b208b5a6-8874-4dda-b24d-bc411ef9874a" />

**Comment by vancura** [comment]
I've broken down all the work from this issue into separate tickets under the new epic #16794 (Global navigation UX). Here's how the topics from this thread connect to the new issues:

- **Overflow / too many items**: #12775 (scrolling, PR in progress), #16800 (fade gradients), #16801 (bottom panel always visible)

- **Pinning / customization**: #16802 (pin items), #16803 (hide/show), #16804 (More button for hidden items)

- **Context menus** (VSCode/JetBrains-style right-click): #16807, #16806

- **Resizable navbar / icon+title vs icon-only**: #16805, #16808 (conditional tooltips)

- **Keyboard shortcuts**: #16809, #16810, #16811

- **Accessibility**: #16812, #16813

- **i18n / RTL**: #16814, #16815

- **Pods naming ambiguity**: #16816

You can always check out the prototype at https://podman-desktop-navbar-proto.vancura.dev. The pain-point breakdown is tracked in #16796.

I am closing this issue since the epic and its sub-issues now cover everything. Thank you all for your input over the past two years. Your feedback has shaped the prototype that has helped get us to where we're now.
